lib/request-base.js

```diff
@@ -130,6 +130,50 @@ RequestBase.prototype.timeout = function timeout(options){
 };
 
 /**
+ * Set number of retry attempts on error.
+ *
+ * Failed requests will be retried 'count' times if timeout or err.code >= 500.
+ *
+ * @param {Number} count
+ * @return {Request} for chaining
+ * @api public
+ */
+
+RequestBase.prototype.retry = function retry(count){
+  this._maxRetries = count || 1;
+  this._retries = 0;
+  return this;
+};
+
+/**
+ * Retry request
+ *
+ * @return {Request} for chaining
+ * @api private
+ */
+
+RequestBase.prototype._retry = function(timeout, responseTimeout) {
+  this.clearTimeout();
+
+  // node
+  if (this.req) {
+    this.set(this.req._headers);
+    delete this.req;
+  }
+
+  this._timeout = timeout;
+  this._responseTimeout = responseTimeout;
+  this._aborted = false;
+  this.timedout = false;
+  delete this._timer;
+  delete this._responseTimeoutTimer;
+
+  this._endCalled = false;
+  this.end(this._callback);
```

Comment:
Yes, I can see how this would be a problem. I'll take a look at refactoring `end()` next.